dev/integration_tests/android_testing/lib/src/matcher.dart

```diff
@@ -0,0 +1,177 @@
+  _AndroidSemanticsMatcher({
+    this.text,
+    this.className,
+    this.id,
+    this.actions,
+    this.rect,
+    this.size,
+    this.isChecked,
+    this.isCheckable,
+    this.isEnabled,
+    this.isEditable,
+    this.isFocusable,
+    this.isFocused,
+    this.isPassword,
+    this.isLongClickable,
+  });
+
+  final String text;
+  final String className;
+  final int id;
+  final List<AndroidSemanticsAction> actions;
+  final Rect rect;
+  final Size size;
+  final bool isChecked;
+  final bool isCheckable;
+  final bool isEditable;
+  final bool isEnabled;
+  final bool isFocusable;
+  final bool isFocused;
+  final bool isPassword;
+  final bool isLongClickable;
+
+  @override
+  Description describe(Description description) {
+    description.add('AndroidSemanticsNode');
+    if (text != null)
+      description.add(' with text: $text');
+    if (className != null)
+      description.add(' with className: $className');
+    if (id != null)
+      description.add(' with id: $id');
+    if (actions != null)
+      description.add(' with actions: $actions');
+    if (rect != null)
+      description.add(' with rect: $rect');
+    if (size != null)
+      description.add(' with size: $size');
+    if (isChecked != null)
+      description.add(' with flag isChecked: $isChecked');
+    if (isEditable != null)
+      description.add(' with flag isEditable: $isEditable');
+    if (isEnabled != null)
+      description.add(' with flag isEnabled: $isEnabled');
+    if (isFocusable != null)
+      description.add(' with flag isFocusable: $isFocusable');
+    if (isFocused != null)
+      description.add(' with flag isFocused: $isFocused');
+    if (isPassword != null)
+      description.add(' with flag isPassword: $isPassword');
+    if (isLongClickable != null)
+      description.add(' with flag isLongClickable: $isLongClickable');
+    return description;
+  }
+
+  @override
+  bool matches(covariant AndroidSemanticsNode item, Map<Object, Object> matchState) {
+    if (text != null && text != item.text)
+      return _failWithMessage('Expected text: $text', matchState);
+    if (className != null && className != item.className)
+      return _failWithMessage('Expected className: $className', matchState);
+    if (id != null && id != item.id)
+      return _failWithMessage('Expected id: $id', matchState);
+    if (rect != null && rect != item.getRect())
+      return _failWithMessage('Expected rect: $rect', matchState);
+    if (size != null && size != item.getSize())
+      return _failWithMessage('Expected size: $size', matchState);
+    if (actions != null) {
+      final List<AndroidSemanticsAction> itemActions = item.getActions();
+      if (actions.length != itemActions.length) {
+        return _failWithMessage('Expected actions: $actions', matchState);
+      }
+      final List<int> usedIds = <int>[];
+      outer: for (int i = 0; i < actions.length; i++) {
+        final AndroidSemanticsAction leftAction = actions[i];
```

Comment:
nit: I was a bit confused by the name, maybe call it expected Action?

Reply:
Removed by below